packages/ripple-binary-codec/src/enums/index.ts

```diff
@@ -1,164 +1,34 @@
+  FieldInstance,
+  Bytes,
+} from './xrpl-definitions-base'
+/**
+ * By default, coreTypes from the `types` folder is where known type definitions are initialized to avoid import cycles.
  */
-class BytesLookup {
-  constructor(types: Record<string, number>, readonly ordinalWidth: number) {
-    Object.entries(types).forEach(([k, v]) => {
-      this[k] = new Bytes(k, v, ordinalWidth)
-      this[v.toString()] = this[k]
-    })
-  }
+const DEFAULT_DEFINITIONS = new XrplDefinitionsBase(enums, {})
 
-  from(value: Bytes | string): Bytes {
-    return value instanceof Bytes ? value : (this[value] as Bytes)
-  }
-
-  fromParser(parser): Bytes {
-    return this.from(parser.readUIntN(this.ordinalWidth).toString())
-  }
-}
+const Type = DEFAULT_DEFINITIONS.type
+const LedgerEntryType = DEFAULT_DEFINITIONS.ledgerEntryType
+const TransactionType = DEFAULT_DEFINITIONS.transactionType
+const TransactionResult = DEFAULT_DEFINITIONS.transactionResult
+const Field = DEFAULT_DEFINITIONS.field
 
 /*
- * type FieldInfo is the type of the objects containing information about each field in definitions.json
- */
-interface FieldInfo {
-  nth: number
-  isVLEncoded: boolean
-  isSerialized: boolean
-  isSigningField: boolean
-  type: string
-}
-
-interface FieldInstance {
-  readonly nth: number
-  readonly isVariableLengthEncoded: boolean
-  readonly isSerialized: boolean
-  readonly isSigningField: boolean
-  readonly type: Bytes
-  readonly ordinal: number
-  readonly name: string
-  readonly header: Buffer
-  readonly associatedType: typeof SerializedType
-}
-
-function buildField([name, info]: [string, FieldInfo]): FieldInstance {
-  const typeOrdinal = enums.TYPES[info.type]
-  const field = fieldHeader(typeOrdinal, info.nth)
-  return {
-    name: name,
-    nth: info.nth,
-    isVariableLengthEncoded: info.isVLEncoded,
-    isSerialized: info.isSerialized,
-    isSigningField: info.isSigningField,
-    ordinal: (typeOrdinal << 16) | info.nth,
-    type: new Bytes(info.type, typeOrdinal, TYPE_WIDTH),
-    header: field,
-    associatedType: SerializedType, // For later assignment in ./types/index.js
-  }
-}
-
-/*
- * @brief: The collection of all fields as defined in definitions.json
+ * @brief: All valid transaction types
  */
-class FieldLookup {
-  constructor(fields: Array<[string, FieldInfo]>) {
-    fields.forEach(([k, v]) => {
-      this[k] = buildField([k, v])
-      this[this[k].ordinal.toString()] = this[k]
-    })
-  }
-
-  fromString(value: string): FieldInstance {
-    return this[value] as FieldInstance
-  }
-}
-
-const Type = new BytesLookup(enums.TYPES, TYPE_WIDTH)
-const LedgerEntryType = new BytesLookup(
-  enums.LEDGER_ENTRY_TYPES,
-  LEDGER_ENTRY_WIDTH,
-)
-const TransactionType = new BytesLookup(
-  enums.TRANSACTION_TYPES,
-  TRANSACTION_TYPE_WIDTH,
-)
-const TransactionResult = new BytesLookup(
-  enums.TRANSACTION_RESULTS,
-  TRANSACTION_RESULT_WIDTH,
-)
-const Field = new FieldLookup(enums.FIELDS as Array<[string, FieldInfo]>)
```

Comment:
All the Field methods are copied unchanged into `field.ts` here: https://github.com/XRPLF/xrpl.js/blob/pass-through-encode/packages/ripple-binary-codec/src/enums/field.ts

(With the last line `const Field` being defined elsewhere in this file (`enums/index.ts`) using `DEFAULT_DEFINITIONS` which is designed to be equivalent to previous behavior.)